src/darwin/frida-helper-backend-glue.m

```diff
@@ -2470,6 +2456,305 @@ static gboolean frida_agent_context_init_functions (FridaAgentContext * self, Gu
 #endif
 }
 
+static gboolean
+frida_find_libc (const GumModuleDetails * details, gpointer user_data)
+{
+  GumAddress * address = user_data;
+
+  if (g_str_has_suffix (details->path, "libSystem.B.dylib"))
+  {
+    *address = details->range->base_address;
+    return FALSE;
+  }
+
+  return TRUE;
+}
+
+static GumAddress
+frida_find_libc_initializer (guint task, GumAddress base)
+{
+  gpointer image;
+  const struct mach_header * header;
+  gconstpointer command;
+  gsize command_index;
+  GumAddress initializer = 0;
+  gsize slide = 0;
+
+  image = gum_darwin_read (task, base, 4096, NULL);
+  header = image;
+
+  if (header->magic == MH_MAGIC)
+    command = image + sizeof (struct mach_header);
+  else
+    command = image + sizeof (struct mach_header_64);
+
+  slide = frida_get_module_slide (command, header->ncmds, base);
+
+  for (command_index = 0;
+       command_index != header->ncmds && initializer == 0;
+       command_index++)
+  {
+    const struct load_command * lc = command;
+
+    if (lc->cmd == LC_SEGMENT || lc->cmd == LC_SEGMENT_64)
```

Comment:
Maybe we can invert this check and do a `goto skip_command;` so we eliminate one level of nesting?